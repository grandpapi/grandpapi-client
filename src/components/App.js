import React from 'react';
import {
  BrowserRouter as Router,
  Route,
  Switch
} from 'react-router-dom';
import Callback from '../containers/auth/Callback';
import Home from '../pages/Home';
import { withSession } from '../containers/auth/withSession';
import Dashboard from '../pages/Dashboard';
import Create from '../pages/Create';
import Docs from '../pages/Docs';
import AboutApp from '../pages/AboutApp';
import PublicApis from '../pages/PublicApis';
import Login from '../pages/Login';
import DatabaseDetail from '../pages/DatabaseDetail';
import { withoutSession } from '../containers/auth/withoutSession';

export default function App() {
  return (
    <Router>
      <Switch>
        <Route exact path="/" component={withoutSession(Home)} />
        <Route exact path="/login" component={withoutSession(Login)} />
        <Route path="/callback" component={Callback} />
        <Route exact path="/about" component={AboutApp} />
        <Route exact path="/docs" component={Docs} />
        <Route exact path="/apis" component={PublicApis} />
        <Route path="/dashboard/:dbName" component={withSession(DatabaseDetail)}/>
        <Route exact path="/dashboard" component={withSession(Dashboard)} />
        <Route exact path="/create" component={withSession(Create)} />
      </Switch>
    </Router>
  );
}
