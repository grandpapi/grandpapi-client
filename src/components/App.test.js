import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

jest.mock('../services/auth.js', () => ({
  handleAuth() {
    return Promise.resolve([]);
  },
  handleCheck() {
    return Promise.resolve([]);
  }
}));

jest.mock('../services/dbMegaNapAPI.js', () => ({
  postDb() {
    return Promise.resolve([]);
  },
  getDbs() {
    return Promise.resolve([]);
  }
}));

jest.mock('../services/modelMegaNapAPI.js', () => ({
  postModel() {
    return Promise.resolve([]);
  },
  getModels() {
    return Promise.resolve([]);
  },
  getAllModels() {
    return Promise.resolve([]);
  }
}));

jest.mock('../services/dataMegaNapAPI.js', () => ({
  postData() {
    return Promise.resolve([]);
  }
}));

describe('App component', () => {
  it('renders App', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
