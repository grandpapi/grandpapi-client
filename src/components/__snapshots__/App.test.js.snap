// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/about"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/docs"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/apis"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(WithSession)",
          "type": [Function],
        }
      }
      path="/dashboard"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(WithSession)",
          "type": [Function],
        }
      }
      exact={true}
      path="/create"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Callback)",
          "type": [Function],
        }
      }
      path="/callback"
    />
  </Switch>
</BrowserRouter>
`;
